### ISSUE TITLE
Add ability to override package manager auto-detection

### ISSUE THREAD

**Issue by wincent** [postmodern/ruby-install#253]
I'm on a corporate-managed OS X machine that happens to have both `brew` and `yum` installed on it, and I am running afoul of [these lines](https://github.com/postmodern/ruby-install/blob/0623212ca5402bae2635f350c4e7099931987c3c/share/ruby-install/util.sh#L6-L12) which auto-detect the package manager to use.

In this case, it finds the corp-managed `yum` first and tries to use it, then blows up. If I change the order if the conditional statements, I can force the use of `brew`, and installation works.

``` diff
diff --git a/share/ruby-install/util.sh b/share/ruby-install/util.sh
index 4ea5b48..8fc8fbd 100644
--- a/share/ruby-install/util.sh
+++ b/share/ruby-install/util.sh
@@ -5,9 +5,9 @@
 #
 if   command -v apt-get >/dev/null; then package_manager="apt"
 elif command -v dnf     >/dev/null; then package_manager="dnf"
+elif command -v brew    >/dev/null; then package_manager="brew"
 elif command -v yum     >/dev/null; then package_manager="yum"
 elif command -v port    >/dev/null; then package_manager="port"
-elif command -v brew    >/dev/null; then package_manager="brew"
 elif command -v pacman  >/dev/null; then package_manager="pacman"
 fi
```

What do you think about changing the order of those lines to prefer `brew` over `yum`? If you think that in itself may just create an issue for some other user, would you accept a PR to make the package manager optionally selectable, say, via an environment variable, so I could invoke `ruby-install` as `PACKAGE_MANAGER=brew ruby-install ruby` or similar?

**Comment by penman** [comment]
I have a similar reason to want to use Homebrew over my system package manager (apt). Short of adding an override, maybe it would be reasonable to move Homebrew to the top of the package manager list, since if somebody has gone to the trouble of installing Homebrew in addition to the system package manager, they probably want to use it?

I agree that a way to override would be best though.

**Comment by havenwood** [comment]
Closed by #189.
